### PR TITLE
feat(organization): migrate organization edit create delete to react query

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-844

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery  preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery  preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Take rest sometimes

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts [hunk 1]
```diff
@@ -61,6 +63,20 @@ type CredentialRequest =
     }
 
 export const organizations = createQueryKeys('organizations', {
+  list: () => ({
+    queryKey: ['organizations'],
+    async queryFn() {
+      const response = await organizationApi.listOrganization()
+      return response.data.results
+    },
+  }),
```
Comment (ctjhoa) [inline libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts:72]:
```suggestion
  list: {
    queryKey: null,
    async queryFn() {
      const response = await organizationApi.listOrganization()
      return response.data.results
    },
  },
```

File: libs/domains/organizations/feature/src/lib/hooks/use-create-organization/use-create-organization.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  organizationId?: string
+}
+
+export function useCreateOrganization({ organizationId }: UseCreateOrganizationProps) {
+  const queryClient = useQueryClient()
+
+  return useMutation(mutations.createOrganization, {
+    onSuccess() {
+      queryClient.invalidateQueries({
+        queryKey: queries.organizations.list().queryKey,
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/hooks/use-create-organization/use-create-organization.tsx:15]:
```suggestion
        queryKey: queries.organizations.list.queryKey,
```

File: libs/domains/organizations/feature/src/lib/hooks/use-organization/use-organization.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+export interface UseOrganizationProps {
+  organizationId: string
+  enabled?: boolean
+}
+
+export function useOrganization({ organizationId, enabled }: UseOrganizationProps) {
+  return useQuery({
+    ...queries.organizations.details({ organizationId }),
+    enabled: enabled,
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/hooks/use-organization/use-organization.tsx:12]:
```suggestion
    enabled,
```

File: libs/pages/layout/src/lib/feature/layout/layout.tsx [hunk 4]
```diff
@@ -36,29 +33,34 @@ export function Layout(props: PropsWithChildren<LayoutProps>) {
...
+    const organizationIds: string[] = []
 
-        for (let i = 0; i < result.length; i++) {
-          const organization = result[i]
-          organizationIds.push(organization.id)
-        }
+    for (let i = 0; i < organizations.length; i++) {
+      const organization = organizations[i]
+      organizationIds.push(organization.id)
+    }
```
Comment (ctjhoa) [inline libs/pages/layout/src/lib/feature/layout/layout.tsx:47]:
```suggestion
    const organizationIds = organizations.map(({ id }) => id)
```

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/layout/src/lib/feature/layout/layout.tsx (hunk 4 above)
Comment (RemiBonnet) [reply libs/pages/layout/src/lib/feature/layout/layout.tsx]: https://github.com/Qovery/console/pull/957/commits/0d149391e88d4eedad1522e4b0075b175900a98f

File: libs/domains/organizations/feature/src/lib/hooks/use-organization/use-organization.tsx (hunk 3 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/hooks/use-organization/use-organization.tsx]: https://github.com/Qovery/console/pull/957/commits/0d149391e88d4eedad1522e4b0075b175900a98f

File: libs/domains/organizations/feature/src/lib/hooks/use-create-organization/use-create-organization.tsx (hunk 2 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/hooks/use-create-organization/use-create-organization.tsx]: https://github.com/Qovery/console/pull/957/commits/0d149391e88d4eedad1522e4b0075b175900a98f

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts (hunk 1 above)
Comment (RemiBonnet) [reply libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts]: https://github.com/Qovery/console/pull/957/commits/0d149391e88d4eedad1522e4b0075b175900a98f

**Pull request merged**